packages/experiments/src/components/Shimmer/Shimmer.base.tsx

```diff
@@ -1,161 +1,136 @@
 import * as React from 'react';
-import { BaseComponent, classNamesFunction } from '../../Utilities';
-import { DefaultPalette, IStyleSet } from '../../Styling';
+import {
+  BaseComponent,
+  classNamesFunction,
+  customizable,
+  DelayedRender
+} from '../../Utilities';
 import {
   IShimmerProps,
   IShimmerStyleProps,
   IShimmerStyles,
-  ShimmerElementType,
-  ICircle,
-  ILine,
-  IGap,
-  ShimmerElementVerticalAlign,
+  IShimmerElement,
 } from './Shimmer.types';
-import { ShimmerLine } from './ShimmerLine/ShimmerLine';
-import { ShimmerGap } from './ShimmerGap/ShimmerGap';
-import { ShimmerCircle } from './ShimmerCircle/ShimmerCircle';
+import { ShimmerElementsGroup } from './ShimmerElementsGroup/ShimmerElementsGroup';
+
+export interface IShimmerState {
+  /**
+   * Flag for knowing when to remove the shimmerWrapper from the DOM.
+   */
+  contentLoaded?: boolean;
+}
 
-const LINE_DEFAULT_HEIGHT = 16;
-const GAP_DEFAULT_HEIGHT = 16;
-const CIRCLE_DEFAULT_HEIGHT = 24;
+const TRANSITION_ANIMATION_INTERVAL = 200; /* ms */
 
 const getClassNames = classNamesFunction<IShimmerStyleProps, IShimmerStyles>();
 
-export class ShimmerBase extends BaseComponent<IShimmerProps, {}> {
+@customizable('Shimmer', ['theme'])
+export class ShimmerBase extends BaseComponent<IShimmerProps, IShimmerState> {
   public static defaultProps: IShimmerProps = {
     isDataLoaded: false,
     isBaseStyle: false
   };
+
   private _classNames: { [key in keyof IShimmerStyles]: string };
+  private _lastTimeoutId: number | undefined;
+
   constructor(props: IShimmerProps) {
     super(props);
+
+    this.state = {
+      contentLoaded: props.isDataLoaded
+    };
+
+    this._warnDeprecations({
+      'isBaseStyle': 'customElementsGroup',
+      'width': 'widthInPercentage or widthInPixel',
+      'lineElements': 'shimmerElements'
+    });
+
+    this._warnMutuallyExclusive({
+      'lineElements': 'shimmerElements',
+      'customElementsGroup': 'lineElements'
+    });
+  }
+
+  public componentWillReceiveProps(nextProps: IShimmerProps): void {
+    const { isDataLoaded } = nextProps;
+
+    if (this._lastTimeoutId !== undefined) {
+      this._async.clearTimeout(this._lastTimeoutId);
+      this._lastTimeoutId = undefined;
+    }
+    if (isDataLoaded) {
+      this._lastTimeoutId = this._async.setTimeout(() => {
+        this.setState({
+          contentLoaded: isDataLoaded
+        });
+        this._lastTimeoutId = undefined;
+      }, TRANSITION_ANIMATION_INTERVAL);
+    } else {
+      this.setState({
+        contentLoaded: isDataLoaded
+      });
+    }
   }
 
   public render(): JSX.Element {
     const {
       getStyles,
```

Comment:
Approved, but `getStyles` is being replaced with `styles` in 6.0. Depending on when you want this merged, before or after 6.0 is merged to master, you may need to replace getStyles in the props with styles. The function `getStyles` within the .styles.ts file can remain `getStyles`.

Reply:
See this commit for details: https://github.com/OfficeDev/office-ui-fabric-react/commit/f226fb413dc5d88a36092b9e70b081313c5da37e